assets/src/latest-stories-block/block/authorSelection.js

```diff
@@ -33,19 +32,34 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import Autocomplete from './autocomplete';
+import { FETCH_AUTHORS_DEBOUNCE } from './constants';
 
 /**
  * Module Constants
  */
 const USERS_LIST_QUERY = {
-  per_page: -1,
+  per_page: 100,
 };
 
+/**
+ * AuthorSelection component. Used for selecting authors of stories.
+ *
+ * @param {Object} props Component props.
+ * @param {Array} props.authors An array of authors objects which are currently selected.
+ * @param props.setAttributes Callable function for saving attribute values.
+ * @return {*} JSX markup.
+ */
 const AuthorSelection = ({ authors, setAttributes }) => {
   const [authorsList, setAuthorsList] = useState([]);
 
+  /**
+   * Returns an array of author objects with author's name as the key. An object will contain 'id' and 'value' field.
+   * 'value' is author's name.
+   *
+   * @return {Array} An array of author objects with author's name as the key.
+   */
   const getAuthorSuggestions = () => {
-    if (undefined !== authorsList && authorsList.length > 0) {
+    if (Array.isArray(authorsList) && authorsList.length > 0) {
```

Comment:
You want to check if `authorList` is defined or not else `Array.isArray()` will throw `Uncaught ReferenceError` for it not being defined.